### PR TITLE
Run sanity tests

### PR DESCRIPTION
**Pull request by ethernetdan**
We should be running sanity tests with each smoke run

### CODE REVIEW

File: installer/tests/sanity/k8s_test.go [hunk 1]
```diff
@@ -93,42 +117,71 @@ func testAllPodsRunning(t *testing.T) {
...
 		if statusCode/100 != 2 {
-			t.Fatalf("expected 200 from log response, got %d", statusCode)
+			return fmt.Errorf("expected 200 from log response, got %d", statusCode)
 		}
-		return
+		return nil
 	}
 
-	t.Errorf("failed to find tectonic-identity pod")
+	return fmt.Errorf("failed to find tectonic-identity pod (found pods in %s: %s)", namespace, names)
```
Comment (ethernetdan) [inline installer/tests/sanity/k8s_test.go]: @Quentin-M @squat any idea why we're not seeing any Pods in the `tectonic-system` namespace during the smoke test?

/cc @ericchiang

**[Smoke Logs](https://jenkins-tectonic-installer.prod.coreos.systems/job/coreos%20-%20tectonic-installer/job/tectonic-installer/job/PR-334/20/console):**
```
[TerraForm: AWS]     	k8s_test.go:133: Failed to get Pod logs with error:  failed to find tectonic-identity pod (found pods in tectonic-system: )
[TerraForm: AWS]     	k8s_test.go:140: Failed to gather logs for tectonic-system/tectonic-identity* in 3m0s
```

**Comment by Quentin-M** [comment]
Timeout is too low? Kubelet has to start and bootkube has to finish before the Tectonic script starts.

**Comment by ethernetdan** [comment]
@Quentin-M I'll set it very high and try again

**Comment by ethernetdan** [comment]
@Quentin-M still failed after 30minutes 😞
**[Smoke Logs](https://jenkins-tectonic-installer.prod.coreos.systems/job/coreos%20-%20tectonic-installer/job/tectonic-installer/job/PR-334/23/console):**
```
[TerraForm: AWS]     	k8s_test.go:140: Failed to gather logs for tectonic-system/tectonic-identity* in 30m0s
```

Any chance Tectonic manifests are not being created? Do I need a specific tfvars configuration for that to happen?

**Comment by Quentin-M** [comment]
@ethernetdan No, the `tectonic.service` should kick in as soon as `bootkube.service` is done. I'd try to pull the logs of both of these services.

File: installer/tests/sanity/k8s_test.go (hunk 1 above)
Comment (sym3tri) [inline installer/tests/sanity/k8s_test.go]: should be there

**Comment by ethernetdan** [comment]
Skipping logging for now - looks like it's an issue with an incorrect pull secret, will fix in another PR.

This adds a few minutes to the test but I think it's worth the verification

File: installer/tests/sanity/k8s_test.go [hunk 2]
```diff
@@ -91,44 +113,75 @@ func testAllPodsRunning(t *testing.T) {
 }
 
 func testLogs(t *testing.T) {
+	// TODO: Diagnose why this fails.
```
Comment (sym3tri) [inline installer/tests/sanity/k8s_test.go]: Sounds flaky. I'd rather we remove flaky tests and add them back later once we figure out the causes.

Comment (ethernetdan) [inline installer/tests/sanity/k8s_test.go]: Right now we are skipping this but it seems to be a misconfigured pull secret, I'll fix in a later PR.

Comment (ethernetdan) [inline installer/tests/sanity/k8s_test.go]: I was able to test it locally but it just doesn't work on jenkins.

File: Jenkinsfile [hunk 3]
```diff
@@ -91,6 +91,10 @@ pipeline {
             trap shutdown EXIT
 
             make apply
+
+            export TEST_KUBECONFIG=${WORKSPACE}/build/${CLUSTER}/generated/auth/kubeconfig
+            export NODE_COUNT=7 # TODO(DG): should be calculated automatically
```
Comment (Quentin-M) [inline Jenkinsfile]: what does that involve? you can easily grep it out of the tfvars file!

File: installer/tests/sanity/k8s_test.go (hunk 2 above)
Comment (Quentin-M) [inline installer/tests/sanity/k8s_test.go]: RBAC?

File: Jenkinsfile (hunk 3 above)
Comment (ethernetdan) [inline Jenkinsfile]: I'll add that for now but I'd like to have a more defined way of doing this eventually

File: installer/tests/sanity/k8s_test.go (hunk 1 above)
Comment (Quentin-M) [inline installer/tests/sanity/k8s_test.go]: The most straight forward way to see that is to `journalctl -u {tectonic,bootkube}` on the the master instance of the lowest Instance ID.

File: Jenkinsfile (hunk 3 above)
Comment (ethernetdan) [inline Jenkinsfile]: Updated.

File: installer/tests/sanity/k8s_test.go (hunk 2 above)
Comment (ethernetdan) [inline installer/tests/sanity/k8s_test.go]: @Quentin-M I think it's just a bad credential in Jenkins:
```
The Secret "coreos-pull-secret" is invalid: data[.dockerconfigjson]: Invalid value: "<secret contents redacted>": invalid character 'e' looking for beginning of value
May 02 22:46:57 ip-10-0-28-10 bash[4033]: [  274.027778] hyperkube[5]: kubectl failed, retrying in 5 seconds
```

Comment (Quentin-M) [inline installer/tests/sanity/k8s_test.go]: Definitely! Your secret is invalid. Your `tectonic_pull_secret_path` file has has to contain something such as:

```
{
  "auths": {
    "quay.io": {
      "auth": "<redacted>",
      "email": ""
    }
  }
}
```

The format changed between the legacy installer and this one.

Comment (ethernetdan) [inline installer/tests/sanity/k8s_test.go]: @Quentin-M Going to enable checks in another PR to give them some soak time and get these checks in.

**Comment by ethernetdan** [comment]
Will do

**Pull request merged**